### PR TITLE
Fixing ELU and SELU GradGrad

### LINKED ISSUE #19333: Bug in EluGradGrad
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes, see below
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary - latest pip tf_nightly
- **TensorFlow version (use command below)**: v1.8.0-1674-gd8fac4cb80 1.9.0-dev20180515
- **Python version**: Python 3.6.3
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: CUDA Version 9.1.85
- **GPU model and memory**: GeForce GTX 970
- **Exact command to reproduce**: See script attached

### Describe the problem

There seems to be an issue with EluGradGrad that I have uncovered. Please see the post here: https://github.com/renmengye/tensorflow-forward-ad/issues/2#issuecomment-389321546

### Source code / logs

Running this script will demonstrate the incorrect values:

```python
import tensorflow as tf
import numpy as np

def fwd_gradients(ys, xs, d_xs):
    dummy = tf.zeros_like(ys)
    g = tf.gradients(ys, xs, grad_ys=dummy, name="gradients")
    return tf.gradients(g, dummy, grad_ys=d_xs, name="jvp")

def my_elu(x):
    return tf.where(x >= 0.0, x, tf.exp(x) - 1.0)

def main():
    print(tf.__version__)

    sess = tf.InteractiveSession()
    init = tf.global_variables_initializer()
    
    # activation = my_elu # Works correctly tf.nn.relu (or any other non-elu activation)
    activation = tf.nn.elu

    x_size = 3
    y_size = x_size

    # Single ELU or RELU op
    X = tf.placeholder(tf.float64, shape=[x_size]) # Input
    Y = activation(X) # Output

    # Define vjp and jvp
    Vx = tf.placeholder(tf.float64, shape=[x_size]) # Jac-vector product input V
    Vy = tf.placeholder(tf.float64, shape=[y_size]) # vector-jac product input V
    jvp = fwd_gradients(Y, X, d_xs=Vx)
    vjp = tf.gradients(Y, X, grad_ys=Vy)

    # Compute jacobians
    x = np.ones(x_size) - 1.5 # Bug only occurs in x < 0 region
    # x = np.random.normal(-1, 1, x_size)
    tf_jac, numeric_jac = tf.test.compute_gradient(X, [x_size], Y, [y_size], x_init_value=x)
    vjp_jac = np.array([sess.run(vjp, feed_dict={X: x, Vy: v})[0] for v in np.identity(y_size)])
    jvp_jac = np.array([sess.run(jvp, feed_dict={X: x, Vx: v})[0] for v in np.identity(x_size)])

    # Print results as maximum absolute error
    print("Numeric jac:", numeric_jac)
    print("jvp jac:", jvp_jac)
    print("tf error:", np.max(np.abs(numeric_jac - tf_jac)))   # ~0.0
    print("vjp error:", np.max(np.abs(numeric_jac - vjp_jac))) # ~0.0
    print("jvp error:", np.max(np.abs(numeric_jac - jvp_jac))) # LARGE! for ELU

    sess.close()

if __name__ == '__main__':
    main()
```

The solution is to edit the implementation of `_EluGradGrad` [here](https://github.com/tensorflow/tensorflow/blob/f318765ad5a50b2fbd7cc08dd4ebc249b3924270/tensorflow/python/ops/nn_grad.py#L364).

I've created a pull request that references this issue.

### PR DESCRIPTION
**Pull request by lawsonfulton**
Changing EluGradGrad and SeluGradGrad to match ReluGradGrad

Fixes #19333

### CODE REVIEW

**Comment by zero-impact** [comment]
I signed it!

File: tensorflow/python/ops/nn_grad.py [hunk 1]
```diff
@@ -364,7 +364,7 @@ def _ReluGrad(op, grad):
 @ops.RegisterGradient("EluGrad")
 def _EluGradGrad(op, grad):
   elu_x = op.inputs[1]
-  return (gen_nn_ops.elu_grad(grad, op.outputs[0]),
+  return (gen_nn_ops.elu_grad(grad, elu_x),
```
Comment (zheng-xq) [inline tensorflow/python/ops/nn_grad.py]: Good catch! 

Do the test here pass both before and after this change, even with gradient_checker? 

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/kernel_tests/relu_op_test.py#L255

If so, we need to patch up the test coverage hole. Thanks!

**Review by zheng-xq — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/nn_grad.py (hunk 1 above)
Comment (zero-impact) [reply tensorflow/python/ops/nn_grad.py]: Thanks for following up. I don't have the time right now, but can make the changes in a week or two.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 33 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 48 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 63 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 78 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by yifeif** [comment]
@zero-impact are you still working on this PR?

**Comment by zero-impact** [comment]
Ostensibly yes, but unfortunately I haven't really put aside the time to
work on it and implement the tests.

If someone else would like to patch it up that's fine by me, but otherwise
I can try and get it closed off in the next few weeks.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by zero-impact** [comment]
Nope sorry. Deadlines on other projects..

Earliest I could do is give it a shot mid October.

**Review by alextp — CHANGES_REQUESTED** [review]
Hey, can you add a gradient checker test for the second derivative, to make it sure that these are correct?

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by harshini-gadige** [comment]
@zero-impact  Any update ?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
It has been 28 days that this pull-request has stalled. Please create a new pull-request with the requested changes.

**Comment by lawsonfulton** [comment]
Looking into this again after forgetting about it and looks like it has been fixed: https://github.com/mark0725/tensorflow/pull/1447/commits/84c5a4551e2e71d854932cb389c359db11cfa2b1
